src/compat/db_json.cpp

```diff
@@ -856,6 +871,12 @@ JSON_VALIDATOR::JSON_VALIDATOR (const JSON_VALIDATOR &copy)
   else
     {
       m_schema_raw = strdup (copy.m_schema_raw);
+#ifdef SERVER_MODE
+      if (m_schema_raw != NULL)
```

Comment:
Do we need to consider this kind of exception while the existing code doesn't do it?

Reply:
In the reference of `strdup()`, it can return 0 when it fails memory allocation.

Reply:
I know it can return 0. What I meant is that if the `strdup` fails, the existing code doesn't seem to work. It means this code doesn't assume it can be 0.
Or, is this allowed to be NULL? or just for safe?